sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -332,12 +422,62 @@ DeviceImage &ProgramManager::getDeviceImage(OSModuleHandle M, KernelSetId KSId,
   return *Img;
 }
 
-void ProgramManager::build(RT::PiProgram Program, const string_class &Options,
-                           std::vector<RT::PiDevice> Devices) {
+static std::vector<RT::PiProgram> getDeviceLibPrograms(
+    const RT::PiContext Context,
+    const std::vector<RT::PiDevice> &Devices,
+    std::map<DeviceLibExt, RT::PiProgram> &CachedLibPrograms) {
+
+  std::vector<RT::PiProgram> Programs;
+
+  // TODO: SYCL compiler should generate a list of required extensions for a
+  // particular program in order to allow us do a more fine-grained check here.
+  // Require *all* possible devicelib extensions for now.
+  std::pair<DeviceLibExt, bool> RequiredDeviceLibExt[] = {
```

Comment:
Please leave a comment about what `bool` stands for: I originally though that it means required/not required, but looking at the code below, it means `FallbackIsLoaded`

Reply:
Looks good to me